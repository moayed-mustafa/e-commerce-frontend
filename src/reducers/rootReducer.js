
import { combineReducers } from 'redux'
import productsReducer from './products'
import cartReducer from './cart'
import wishlistReducer from './wishlist'
const rootReducer = combineReducers({products:productsReducer,
    cart: cartReducer,
    wishlist: wishlistReducer
})
export default rootReducer;

