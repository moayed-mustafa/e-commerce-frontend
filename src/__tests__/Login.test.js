
import React from "react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";

import rootReducer from '../reducers/rootReducer';
import Login from "../Login";
import UserContext from '../userContext'



let store = createStore(rootReducer);

let current_user= {
    username: "Loginer", _token:
        "123"
}

function checkUser(user) {
    if (!user || !user._token) {
        localStorage.removeItem("user");

    }else {
            localStorage.setItem("user",JSON.stringify(user))
    }

}

describe('<Login />', () => {

    it('passes smoke test', function() {
        render((
            <UserContext.Provider value={{current_user, set_current_user:checkUser}}>
                <Provider store={store}>
                    <Login />
                </Provider>

            </UserContext.Provider>
        ));
    });

    it('passes snapshot test', function() {
        const { asFragment } = render((
            <UserContext.Provider value={{current_user, set_current_user:checkUser}}>
                <Provider store={store}>
                    <Login />
                </Provider>
            </UserContext.Provider>
        ));
        expect( asFragment() ).toMatchSnapshot();
    });

});
