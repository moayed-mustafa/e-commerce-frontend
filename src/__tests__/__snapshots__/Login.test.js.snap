// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> passes snapshot test 1`] = `
<DocumentFragment>
  <div
    class="form-div"
  >
    <h3>
      Login Form
    </h3>
    <form
      class="form"
    >
      <div
        class="form-group form-group"
      >
        <label
          class="ml-1"
          for="username"
        >
          Username:
        </label>
        <input
          class="username form-control"
          name="username"
          type="text"
          value=""
        />
      </div>
      <div
        class="form-group form-group"
      >
        <label
          class="ml-1"
          for="password"
        >
          Password:
        </label>
        <input
          class="password form-control"
          name="password"
          type="password"
          value=""
        />
      </div>
      <button
        class="li-btn btn btn-secondary"
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
</DocumentFragment>
`;
